README.md

```diff
@@ -3,26 +3,55 @@ Arduino MCP2515 CAN interface library
 
 
 <br>
-CAN-BUS is a common industrial bus because of its long travel distance, medium communication speed and high reliability. It is commonly found on modern machine tools and as an automotive diagnostic bus. This CAN-BUS Shield adopts MCP2515 CAN Bus controller with SPI interface and MCP2551 CAN transceiver to give your Arduino/Seeeduino CAN-BUS capibility. With an OBD-II converter cable added on and the OBD-II library imported, you are ready to build an onboard diagnostic device or data logger.
+CAN-BUS is a common industrial bus because of its long travel distance, medium communication speed and high reliability. It is commonly found on modern machine tools and as an automotive diagnostic bus. This CAN-BUS Shield gives your Arduino/Seeeduino CAN-BUS capibility. With an OBD-II converter cable added on and the OBD-II library imported, you are ready to build an onboard diagnostic device or data logger.
 
 - Implements CAN V2.0B at up to 1 Mb/s
 - SPI Interface up to 10 MHz
 - Standard (11 bit) and extended (29 bit) data and remote frames
 - Two receive buffers with prioritized message storage
 
-## Library Installation
+**Contents:**
+* [Hardware](#hardware)
+   * [CAN Shield](#can-shield)
+   * [Do It Yourself](#do-it-yourself)
+* [Software Usage](#software-usage)
+   * [Library Installation](#library-installation)
+   * [Initialization](#initialization)
+   * [Frame data format](#frame-data-format)
+   * [Send Data](#send-data)
+   * [Receive Data](#receive-data)
+   * [Set Receive Mask and Filter](#set-receive-mask-and-filter)
+   * [Examples](#examples)
 
-1. Download the ZIP file from https://github.com/autowp/arduino-mcp2515/archive/master.zip
-2. From the Arduino IDE: Sketch -> Include Library... -> Add .ZIP Library...
-3. Restart the Arduino IDE to see the new "mcp2515" library with examples
+# Hardware:
+
+## CAN Shield
+
+The following code samples uses the CAN-BUS Shield, wired up as shown:
 
+![MCP2515 CAN-Shield wiring](examples/wiring.png)
 
+## Do It Yourself
 
-# Usage:
+If you want to make your own CAN board for under $10, you can achieve that with something like this:
 
+![MCP2515 with MCP2551 wiring](examples/wiring-diy.png)
 
+Component References:
```

Comment:
How about?
[TJA1054](https://www.nxp.com/docs/en/data-sheet/TJA1055.pdf) Fault-tolerant low speed CAN Transceiver. Mostly used in vehicles.